packages/typespec-client-generator-core/design-docs/flags.md

```diff
@@ -0,0 +1,35 @@
+# TCGC `tspconfig.yaml` flags
+
+## 1. `generate-protocol-methods`
+
+When set to `true`, the emitter will generate low-level protocol methods for each service operation if `@protocolAPI` is not set for an operation. Default value is `true`.
+
+## 2. `generate-convenience-methods`
+
+When set to `true`, the emitter will generate low-level protocol methods for each service operation if `@convenientAPI` is not set for an operation. Default value is `true`.
+
+## 3. `package-name`
+
+The `package-name` override you would like for your generated library. By default individual language emitters should calculate the package name for their library from the `SdkPackage.namespaces` that `tcgc` returns to them.
```

Comment:
Feels like we may want to kill it?

Reply:
we have been talking about coalescing to `package-name` across the js, csharp, and python libraries (java falls out of this). Would `package-name` live in tcgc or would each language emitter have their own flag called `package-name`? Now that I'm writing it out, I think it is the latter.

Reply:
this flag has been removed.